packages/mdx/src/mdx-client/slideshow.tsx

```diff
@@ -32,12 +39,26 @@ export function Slideshow({
     (child: any) => child.props?.children
   )
 
+  const maxSteps = editorSteps.length - 1;
+
+  // This hook will prevent the slide from being changed via the initialSlideIndex prop after render
+  const initialSlideValue = useInitialState(initialSlideIndex);
+
+  // Make sure the initial slide is not configured out of bounds
+  const initialSlide = initialSlideValue > maxSteps ? maxSteps : initialSlideValue
+
   const [state, setState] = React.useState({
-    stepIndex: 0,
-    step: editorSteps[0],
+    stepIndex: initialSlide,
+    step: editorSteps[initialSlide],
   })
   const tab = state.step
 
+  // Run any time our Slideshow state changes
+  React.useEffect(() => {
+    // Return our state object to the Slideshow onChange function
+    onSlideshowChange(state);
```

Comment:
as you said, it's better to call back only with the index, it helps to avoid breaking changes in the future

Reply:
maybe an object with just the index (`{index: state.stepIndex}`) so it can be extended later if needed

Reply:
Done. And since we are limiting the scope of what we expose internally, I adjusted the dependency array as well so we only fire the onChange when the slide changes.